apps/api/src/common/scalars/date.scalar.ts

```diff
@@ -7,12 +8,18 @@ export class DateScalar implements CustomScalar<string, Date> {
   description =
     'Date custom scalar type, for (de)serializing Date values from/to ISO8601 representation.'
 
-  parseValue(value: string): Date {
-    return new Date(value) // value from the client
+  parseValue(value: unknown): Date {
+    if (typeof value === 'string') {
+      return new Date(value)
+    }
+    throw new ValidationError('Cannot parse value for Date scalar. Value must be string.')
   }
```

Comment:
```suggestion
  parseValue(value: unknown): Date {
    if (typeof value === 'string') {
      const date = new Date(value)
      if(isNan(date)) throw new SomeError()
      return date
    }
    throw new ValidationError('Cannot parse value for Date scalar. Value must be string.')
  }
```

Since `new Date(randomString)` might be `invalid Date`